src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -211,21 +211,17 @@ ParsedQuery Visitor::visit(Parser::QueryContext* ctx) {
 
 // ____________________________________________________________________________________
 ParsedQuery Visitor::visit(Parser::QueryOrUpdateContext* ctx) {
-  if (ctx->update()) {
+  if (ctx->update() && !ctx->update()->update1()) {
     // An empty query currently matches the `update()` rule. We handle this
     // case manually to get a better error message. If an update query doesn't
     // have an `update1()`, then it consists of a (possibly empty) prologue, but
     // has not actual content, see the grammar in `SparqlAutomatic.g4` for
     // details.
-    if (!ctx->update()->update1()) {
-      reportError(ctx->update(),
-                  "Empty query (this includes queries that only consist "
-                  "of comments or prefix declarations).");
-    }
-    reportNotSupported(ctx->update(), "SPARQL 1.1 Update is");
-  } else {
-    return visit(ctx->query());
+    reportError(ctx->update(),
+                "Empty query (this includes queries that only consist "
+                "of comments or prefix declarations).");
   }
+  return visitAlternative<ParsedQuery>(ctx->query(), ctx->update());
```

Comment:
Am I seeing this correctly that 
the UPDATEs will now throw inside the Server.cpp (just for clarity, nothing to do here)

Reply:
Yes that's correct. In Sever.cpp the path for processing queries checks that it actually got a query. The update path throws an unsupported exception.